src/shared/components/MapBlock.js

```diff
@@ -5,84 +5,86 @@ import controllable from 'react-controllables';
 import GoogleMap from 'google-map-react';
 import Marker from './Marker.js';
 
-class MapBlock extends Component {	  	
-
-  	constructor(props) {
-    	super(props);
-
-    	this.shouldComponentUpdate = shouldPureComponentUpdate;
-
-    	this._onBoundsChange = (center, zoom, bounds, marginBounds) => {  		
-		    this.props.onCenterChange(center);
-		    this.props.onZoomChange(zoom);	    
-	  	}
-
-	  	this._onChildClick = (key, childProps) => {
-	  		this.props.onCenterChange([childProps.lat, childProps.lng]);
-		}
-
-		this._onChildMouseEnter = (key, childProps) => {
-			
-		}	
-
-		this._onChildMouseLeave = (/* key, childProps */) => {
-			
-		}
-  	}  	
-
-	render() {
-		const markers = this.props.markers
-			.map(marker => {
-				const {id, ...coords} = marker;
-				
-				return(
-					<Marker
-						key={id}
-						text={id}
-						{...coords} />
-				);
-			});
-
-		return (
-			<section style={{ width: '100%', height: '500px' }}>
-				<h4>coords: {this.props.center}</h4>
-				<GoogleMap
-					center={this.props.center}
-					zoom={this.props.zoom}
-					onBoundsChange={this._onBoundsChange}
-					onChildClick={this._onChildClick}
-					onChildMouseEnter={this._onChildMouseEnter}
-					onChildMouseLeave={this._onChildMouseLeave}
-					hoverDistance={20}>
-					{markers}
-				</GoogleMap>
-			</section>
-		);
-	}
+class MapBlock extends Component {
+  constructor(props) {
+    super(props);
 
+    this.shouldComponentUpdate = shouldPureComponentUpdate;
+
+    this.onBoundsChange = (center, zoom, bounds, marginBounds) => {
```

Comment:
You can extract this.XXX as members of MapBlock class like below:

`
class MapBlock extends Component {
  onBoundsChange = (...) => {}
  onChildClick = (...) => {}
...
}
`